plugins/org.jkiss.dbeaver.ext.firebird/src/org/jkiss/dbeaver/ext/firebird/model/FireBirdProcedure.java

```diff
@@ -144,7 +143,7 @@ public void loadProcedureColumns(DBRProgressMonitor monitor) throws DBException
                         }
 
                         boolean notNull = JDBCUtils.safeGetBoolean(dbResult, "NOT_NULL");
-                        int scale = JDBCUtils.safeGetInt(dbResult, JDBCConstants.SCALE);
+                        int scale = Math.abs(JDBCUtils.safeGetInt(dbResult, JDBCConstants.SCALE)); // For some reason, FireBird returns the negative value in the scale field.
```

Comment:
This behavior is described in the linked documentation:

> The **negative** number that specifies the scale for DECIMAL and NUMERIC columns—the number of digits after the decimal point

So, since it's expected, maybe use a negation instead of `abs`-ing it?